src/commands/dev/index.ts

```diff
@@ -432,10 +434,11 @@ export default class Dev extends BaseCommand {
           name: 'port',
           message: `Trying to listen on port ${port}, but something is already using it. What port would you like us to run the API gateway on (you can use the '--port' flag to skip this message in the future)?`,
           validate: (value) => {
-            if (new RegExp('^[1-9]+\\d*$').test(value)) {
-              return true;
+            const port = Number.parseInt(value);
+            if (Number.isNaN(port) || port < 1024 || port > 65535) {
```

Comment:
I find this a little weird since `--port=443` and `--port=80` should be valid. Also I don't have any of these restrictions in wsl2.

Reply:
That's fair, I didn't add exceptions for 443/80 because those are the defaults and presumably you'd only use `--port` if you wanted something that wasn't the default. Can change it to allow those ports though, the downside is we can't use oclif's min/max in the flag itself.

The port restriction that Mary originally noticed is on the browsers side though, not docker - if you run `architect dev --port=101` for example it should work, but Chrome or Firefox will likely not allow you to connect because it's in their list of ports that they don't allow.

I am not 100% sure if all browsers maintain the same exact list of these ports so I didn't want to necessarily hardcode a list of ports we'd warn users about, and rather have them select something outside of the well-known port range which has plenty of reserved ports

Reply:
example - building and running works

<img width="847" alt="image" src="https://user-images.githubusercontent.com/1406391/232894776-00ff80e7-7120-4206-b86d-d98b1009b8d9.png">

but the browser is sad
<img width="1182" alt="image" src="https://user-images.githubusercontent.com/1406391/232894680-f23d7140-1e7f-41b1-a338-e76445029c6d.png">

Reply:
Fair enough. Not super worried about it, but prob worth at least adding 80/443 to the whitelist